<launch>
    <arg name="marker_size"          default="16.5" />
    <arg name="max_new_marker_error" default="0.05" />
    <arg name="max_track_error"      default="0.05" />

    <arg name="cam_image_topic"      default="/usb_cam/image_raw" />
    <arg name="cam_info_topic"       default="/usb_cam/camera_info" />
    <arg name="output_frame"         default="/usb_cam" />

    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
        <param name="video_device" value="/dev/video0" />
        <param name="image_width" value="1280" />
        <param name="image_height" value="720" />
        <param name="pixel_format" value="yuyv" />
        <param name="camera_frame_id" value="usb_cam" />
        <param name="io_method" value="mmap"/>
    </node>
<!-- copied from  ar_track.launch from lab4 cam package-->
    <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
        <param name="marker_size"           type="double" value="$(arg marker_size)" />
        <param name="max_new_marker_error"  type="double" value="$(arg max_new_marker_error)" />
        <param name="max_track_error"       type="double" value="$(arg max_track_error)" />
        <param name="output_frame"          type="string" value="$(arg output_frame)" />

        <remap from="camera_image"  to="$(arg cam_image_topic)" />
        <remap from="camera_info"   to="$(arg cam_info_topic)" />
    </node>

    <node pkg="intera_interface" name="enable_robot" type="enable_robot.py" args="-e" output="screen">
    </node>

    <node pkg="intera_interface" name="joint_trajectory_action_server" type="joint_trajectory_action_server.py" output="screen">
    </node>

    <include file="$(find sawyer_moveit_config)/launch/sawyer_moveit.launch" >
        <arg name= "electric_gripper" value="false"/>
    </include>

    <!-- <node pkg="move_grip" name="path_test" type="path_test.py" args="sawyer" output="screen">
    </node> -->
    <node pkg="robotiq_vacuum_grippers_control" name="RobotiqVacuumGrippersRtuNode" type="RobotiqVacuumGrippersRtuNode.py" args="/dev/ttyUSB0" output="screen">
    </node> 

    <node pkg="move_grip" name="ar_tag_publisher" type="ar_tag_publisher.py" output="screen">
    </node> 
</launch>
